src/assembler.jl

```diff
@@ -1,24 +1,31 @@
-struct Assembler{T}
+abstract type AbstractSparseAssembler end
+abstract type AbstractSparseCSCAssembler <: AbstractSparseAssembler end
+
+"""
+This assembler creates a COO (**coo**rdinate format) representation of a sparse matrix during
+assembly and converts it into a `SparseMatrixCSC{T}` on finalization.
+"""
+struct COOAssembler{T}
     I::Vector{Int}
     J::Vector{Int}
     V::Vector{T}
 end
 
-Assembler(N) = Assembler(Float64, N)
+COOAssembler(N) = COOAssembler(Float64, N)
```

Comment:
We are adding defaults both here and for `start_assemble`, I guess it suffices to add this for `start_assemble`? Alternatively, should we not have the `start_assemble` function for this case, and rather have to explicityly define a `COOAssembler` when required? Seems reasonable to me to request this type explicitly in those special cases?

Reply:
I think from the interface point of view the start_assemble makes more sense here, but I do not have a strong opinion. @fredrikekre ?